knimin/templates/pm_extract_plate.html

```diff
@@ -0,0 +1,186 @@
+        '<div class="modal-dialog" role="document">' +
+          '<div class="modal-content">' +
+            '<div class="modal-header">' +
+              '<h4 class="modal-title" id="addPlateModalLabel">Add plate</h4>' +
+              '<button type="button" class="close" data-dismiss="modal" aria-label="Close">' +
+                '<span aria-hidden="true">&times;</span>' +
+              '</button>' +
+            '</div>' +
+            '<div class="modal-body">' +
+              '<div class="input-group">' +
+                '<select id="plate-select">' +
+                  '<option selected value="-1">Choose a plate</option>' +
+                '</select>' +
+              '</div>' +
+            '</div>' +
+            '<div class="modal-footer">' +
+              '<button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>' +
+              '<button type="button" class="btn btn-primary" id="add-plate-btn" disabled>Add</button>' +
+            '</div>' +
+          '</div>' +
+        '</div>' +
+      '</div>').appendTo($('body'));
+
+    // Add
+    $.each(plates, function(idx, plate) {
+      $('<option>').attr('value', plate[0]).attr('pm-name', plate[1]).html('(ID: ' + plate[0] + ') ' + plate[1]).appendTo($("#plate-select"));
+    });
+
+    // Remove the plate that is already selected
+    $("#plate-select option[value={{plate_id}}]").remove();
+
+    // Attach handler to the change even of the select
+    $("#plate-select").change(function(e) {
+      $("#add-plate-btn").prop("disabled", this.value === "-1");
+    });
+
+    $("#add-plate-btn").click(function (e) {
+      var plateId = $("#plate-select").val();
+      var plateName = $("#plate-select option[value=" + plateId + "]").attr('pm-name');
+
+      // Add the new plate to the list
+      var $li = $('<li>').addClass('list-group-item justify-content-between').attr('id', 'li-plate-' + plateId).attr('pm-id', plateId).html('(ID: ' + plateId + ') ' + plateName).appendTo($("#lu-plate-list"));
+      var $btn = $('<button>').addClass('btn-danger glyphicon glyphicon-remove pull-right').html(" ").appendTo($li);
+      $btn.attr('pm-name', plateName).attr('pm-id', plateId);
+      $btn.click(function (e) {
+        // Add this plate back to the select
+        $("#plate-select").append($('<option>', {value: $(this).attr('pm-id'),
+                                                 text: '(ID: ' + $(this).attr('pm-id') + ') ' + $(this).attr('pm-name'),
+                                                 "pm-name": $(this).attr('pm-name')}));
+        $li.remove();
+      });
+
+      // Set the selected value to -1
+      $("#plate-select").val("-1");
+      // Delete the selected plate from the option list
+      $("#plate-select option[value=" + plateId + "]").remove();
+      // Hide the modal
+      $('#addPlateModal').modal('hide');
```

Comment:
Minor but addPlateModal is the only camelCased identifier, maybe for consistency worth changing to add-plate-modal